autoinstrumentation/php/prepare_files_for_docker_image.sh

```diff
@@ -0,0 +1,187 @@
+                ;;
+        esac
+        shift
+    done
+
+    if [ -z "${opentelemetry_extension_version}" ] ; then
+        echo "<opentelemetry extension version> argument is missing"
+        show_help
+        exit 1
+    fi
+    if [ -z "${destination_directory}" ] ; then
+        echo "<destination directory> argument is missing"
+        show_help
+        exit 1
+    fi
+}
+
+ensure_dir_exists_and_empty() {
+    local dir_to_clean="${1:?}"
+
+    if [ -d "${dir_to_clean}" ]; then
+        rm -rf "${dir_to_clean}"
+        if [ -d "${dir_to_clean}" ]; then
+            echo "Directory ${dir_to_clean} still exists. Directory content:"
+            ls -l "${dir_to_clean}"
+            exit 1
+        fi
+    else
+        mkdir -p "${dir_to_clean}"        
+    fi
+}
+
+build_native_binaries_for_PHP_version_libc_variant() {
+    local PHP_version="${1:?}"
+    local libc_variant="${2:?}"
+    local dest_dir_for_current_args
+    dest_dir_for_current_args="${destination_directory}/native_binaries/PHP_${PHP_version}_${libc_variant}"
+
+    echo "Building extension binaries for PHP version: ${PHP_version} and libc variant: ${libc_variant} to ${dest_dir_for_current_args} ..."
+
+    ensure_dir_exists_and_empty "${dest_dir_for_current_args}"
+
+    local PHP_docker_image="php:${PHP_version}-cli"
+    local install_compiler_command=""
+    case "${libc_variant}" in
+        glibc)
+            ;;
+        musl)
+            PHP_docker_image="${PHP_docker_image}-alpine"
+            install_compiler_command="&& apk update && apk add autoconf build-base"
+            ;;
+        *)
+            echo "Unexpected libc variant: ${libc_variant}"
+            exit 1
+            ;;
+    esac
+
+    local current_user_id
+    current_user_id="$(id -u)"
+    local current_user_group_id
+    current_user_group_id="$(id -g)"
+    docker run --rm \
+        -v "${dest_dir_for_current_args}:/dest_dir" \
+        ${PHP_docker_image} sh -c "\
+        mkdir -p /app && cd /app \
+        ${install_compiler_command} \
+        && pecl install opentelemetry-${opentelemetry_extension_version} \
+        && cp /usr/local/lib/php/extensions/no-debug-non-zts-*/opentelemetry.so /dest_dir/ \
+        && chown -R ${current_user_id}:${current_user_group_id} /dest_dir/"
+
+    echo "Built extension binaries for PHP version: ${PHP_version} and libc variant: ${libc_variant}"
+}
+
+build_native_binaries() {
+    echo "Building extension binaries..."
+
+    for PHP_version in "${PHP_versions[@]}" ; do
+        for libc_variant in "${libc_variants[@]}" ; do
+            build_native_binaries_for_PHP_version_libc_variant "${PHP_version}" "${libc_variant}"
+        done
+    done
+
+    echo "Built extension binaries"
+}
+
+select_composer_json_for_PHP_version() {
+    local PHP_version="${1:?}"
+    case "${PHP_version}" in
+        8.0|8.1|8.2)
+            echo "composer_PHP_${PHP_version}.json"
+            ;;
+        8.3)
+            echo "composer_PHP_8.2.json"
```

Comment:
Just to clarify, its correct we want the `8.2` filein case of `8.3`?

Reply:
Yes, that is on purpose. Should I add a comment to that code line clarifying that?

The reason for having separate `composer_PHP_8.1.json` and `composer_PHP_8.2.json` is because [`open-telemetry/opentelemetry-auto-pdo` auto-instrumentation is supported only for PHP 8.2 and later](https://packagist.org/packages/open-telemetry/opentelemetry-auto-pdo).

Reply:
This relates to which PHP versions can/cannot observe internal functions, which arrived in 8.2, so it's really `<8.2` and `>=8.2`
I think that needs a comment or improved file names.

Reply:
Please add the comment, and then we should be good to merge.